include/zephyr/input/input_pat912x.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_INPUT_PAT912X_H_
+#define ZEPHYR_INCLUDE_INPUT_PAT912X_H_
+
+/**
+ * @brief Set resolution on a pat912x device
+ *
+ * @param dev pat912x device.
+ * @param res_x_cpi CPI resolution for the X axis, 0 to 1275, -1 to keep the
+ * current value.
+ * @param res_y_cpi CPI resolution for the Y axis, 0 to 1275, -1 to keep the
+ * current value.
+ */
+int pat912x_set_resolution(const struct device *dev,
+			   int16_t res_x_cpi, int16_t res_y_cpi);
```

Comment:
Just for my own curiosity sake: is there any guideline on when to put this kind of functionality into public facing headers vs. using sensor attributes? Personally I feel that a lot of context is lost when using the sensor api for that (var names and types).

Reply:
Yeah well right now it's simple, there's no sensor attributes because it's not a sensor, in fact there's no API at all for input devices :-) so everything is public header.

This very one though stats to make me think we may adopt an attribute API down the road, so my current thinking is: let's get some more drivers in and see where it goes, then we can design something that works for the drivers we have rather than some hypothetical ones.

Reply:
+1, as more drivers exist this could be refactored into a common public API